charts/network-agent/templates/daemonset.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "hny-network-agent.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "hny-network-agent.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "hny-network-agent.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "hny-network-agent.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ include "hny-network-agent.fullname" . }}
+      hostNetwork: true
+      containers:
+        - name: hny-network-agent
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: HONEYCOMB_API_KEY
+              valueFrom:
+                secretKeyRef:
+                  {{- if .Values.honeycomb.existingSecret }}
+                  name: {{ .Values.honeycomb.existingSecret }}
+                  key: {{ .Values.honeycomb.existingSecretKey }}
+                  {{- else }}
+                  name: {{ include "hny-network-agent.fullname" . }}
+                  key: apiKey
+                  {{- end }}
+            - name: HONEYCOMB_API_ENDPOINT
+              value: {{ .Values.honeycomb.endpoint }}
+            - name: HONEYCOMB_DATASET
+              value: {{ .Values.honeycomb.dataset }}
+            - name: HONEYCOMB_STATS_DATASET
+              value: {{ .Values.honeycomb.statsDataset }}
+          securityContext:
+            capabilities:
+              add:
+                - NET_RAW
```

Comment:
Not a blocker for this PR, but It would be a good idea to add a section in the README detailing the permissions/RBAC this chart sets up.

Reply:
We don't have a section in the main repo that describes required permissions yet. We've got the following ticket to track document them, and I've added a note to say we should update the helm chart once we've got it
- https://github.com/honeycombio/honeycomb-network-agent/issues/167